golink.go

```diff
@@ -527,6 +527,15 @@ func serveGo(w http.ResponseWriter, r *http.Request) {
 	}
 
 	link, err := db.Load(short)
+	if errors.Is(err, fs.ErrNotExist) {
+		// Trim common punctuation from the end and try again.
+		// This catches auto-linking and copy/paste issues that include punctuation.
+		if s := strings.TrimRight(short, ".,()[]{}"); short != s {
```

Comment:
Even though it's valid, it might be worth taking `?` too, since such a URL's query is empty anyway at that point (the same argument could apply to `.`, since it's valid in paths). YMMV.

Reply:
hmm... I tried adding `?`, but turns out it's not really needed.  Since this is based solely on the `r.URL.Path` value, it seems that the only way a question mark could appear is if it was url encoded as `%3F`.  But then we never url-decode, so it wouldn't actually get stripped off.  Since we're just trying to handle accidental punctuation, I don't think we need to worry about `?`, since it would get stripped already by the URL parsing.